### PR TITLE
Start Gen9 Hold Effects

### PR DESCRIPTION
**Pull request by ghoulslash**
Adds the following hold effects:
- [x] HOLD_EFFECT_ABILITY_SHIELD
- [x] HOLD_EFFECT_CLEAR_AMULET
- [x] HOLD_EFFECT_PUNCHING_GLOVE
- [x] HOLD_EFFECT_COVERT_CLOAK
- [x] HOLD_EFFECT_LOADED_DICE

The PR also adds a few RecordItemEffect and RecordAbilityBattle calls.

HOLD_EFFECT_MIRROR_HERB I will leave for another PR when more information about edge cases etc, comes out. Information about items taken from bulbapedia and serebii

Questions to handle down the road:
- Does Ability Shield block your own Trace?

Note: I am unable to build pokeemerald these days somehow, so have not tested any. Most are just analogs to existing abilities, though

### CODE REVIEW

File: src/battle_util.c [hunk 1]
```diff
@@ -8011,7 +8028,8 @@ bool32 IsMoveMakingContact(u16 move, u8 battlerAtk)
         else
             return FALSE;
     }
-    else if (GetBattlerAbility(battlerAtk) == ABILITY_LONG_REACH)
+    else if (GetBattlerAbility(battlerAtk) == ABILITY_LONG_REACH
+        || GetBattlerHoldEffect(battlerAtk, TRUE) == HOLD_EFFECT_PUNCHING_GLOVE)
```
Comment (AsparagusEduardo) [inline src/battle_util.c:8032]: `GetBattlerHoldEffect` is being checked twice, gotta save it in a variable.

File: src/battle_util.c [hunk 2]
```diff
@@ -3871,6 +3872,10 @@ u8 AtkCanceller_UnableToUseMove(void)
                 {
                     SetRandomMultiHitCounter();
                 }
+                
+                if (gMultiHitCounter < 4 && GetBattlerHoldEffect(gBattlerAttacker, TRUE) == HOLD_EFFECT_LOADED_DICE)
+                    gMultiHitCounter = 4;
```
Comment (AsparagusEduardo) [inline src/battle_util.c:3877]: Loaded dice doesn't set the counter to 4, it's meant to be *at least* 4, so this should go in ``SetRandomMultiHitCounter`.

Here's the detailed info from the Smogon forums: https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/page-9#post-9400247

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: src/battle_util.c (hunk 1 above)
Comment (ghoulslash) [reply src/battle_util.c:8032]: pretty sure the compiler would optimize this

Comment (AsparagusEduardo) [reply src/battle_util.c:8032]: you sure? I wouldn't think that it would keep that in memory

Comment (ghoulslash) [reply src/battle_util.c]: I checked the compiler explorer and you're right. Just fixed this and loaded dice.

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**